src/storage/gc_worker.rs

```diff
@@ -212,25 +255,104 @@ impl<E: Engine> GCRunner<E> {
         );
         Ok(())
     }
-}
 
-impl<E: Engine> Runnable<GCTask> for GCRunner<E> {
-    fn run(&mut self, mut task: GCTask) {
-        GC_GCTASK_COUNTER.inc();
+    fn unsafe_destroy_range(&self, _: &Context, start_key: &Key, end_key: &Key) -> Result<()> {
+        info!(
+            "destroying range start_key: {}, end_key: {}",
+            start_key, end_key
+        );
+
+        // TODO: Refine usage of errors
+
+        let local_storage = self.local_storage.as_ref().ok_or_else(|| {
+            let e: Error = box_err!("destroy range not supported: local_storage not set");
+            warn!("destroy range failed: {:?}", &e);
+            e
+        })?;
+
+        // Convert keys to RocksDB layer form
+        // TODO: Logic coupled with raftstore's implementation. Maybe better design is to do it in
+        // somewhere of the same layer with apply_worker.
+        let start_data_key = keys::data_key(start_key.as_encoded());
+        let end_data_key = keys::data_end_key(end_key.as_encoded());
+
+        let cfs = &[CF_LOCK, CF_DEFAULT, CF_WRITE];
+
+        // First, call delete_files_in_range to free as much disk space as possible
+        // TODO: Will LOCK_CF cause problem here?
```

Comment:
This range will not be read after table has been truncated and passed lifetime, and gc tasks for this range will not concurrently run with destroy range, so i don't think there is problem.
@disksing